include/arch/arm/aarch64/arch_inlines.h

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_ARCH_ARM_AARCH64_ARCH_INLINES_H
+#define ZEPHYR_INCLUDE_ARCH_ARM_AARCH64_ARCH_INLINES_H
+
+#ifndef _ASMLANGUAGE
+
+#include <kernel_structs.h>
+#include <arch/cpu.h>
+
+static ALWAYS_INLINE _cpu_t *arch_curr_cpu(void)
+{
+#ifdef CONFIG_SMP
+	uint64_t core;
+
+	/* Note: Only support one Cluster */
+	__asm__ volatile("mrs %0, mpidr_el1" : "=r" (core));
+	core = core & 0xFF;
```

Comment:
Can re-use the macros available for GET_CPUID

Reply:
I use MPIDR_TO_CORE